src/main/java/com/exadel/movie_new/controller/MovieController.java

```diff
@@ -1,48 +1,103 @@
 package com.exadel.movie_new.controller;
 
+import com.exadel.movie_new.exception.FilesException;
+import com.exadel.movie_new.exception.GeneralException;
+import com.exadel.movie_new.exception.IdNotValidException;
+import com.exadel.movie_new.exception.MovieNotFoundException;
 import com.exadel.movie_new.model.Movie;
 import com.exadel.movie_new.service.MovieService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
+import static org.apache.commons.lang3.StringUtils.*;
 
-@RestController @RequestMapping("/movies")
+
+@RestController
+@RequestMapping("/movies")
 public class MovieController {
 
-    private MovieService movieService;
+    private final MovieService movieService;
+
 
     public MovieController(MovieService movieService) {
-        this.movieService = movieService;
+        this.movieService =   movieService;
     }
 
 
-    @PostMapping()
-    public Movie addMovie(@RequestBody Movie movie) {
-
-        return movieService.addMovie(movie);
+    @PostMapping
+    public ResponseEntity<Movie> addMovie(@RequestBody Movie movie) {
+        if (!isNumeric(movie.getId())) {
+            throw new IdNotValidException("could not add a movie without valid id");
+        }
+        return new ResponseEntity<>(movieService.addMovie(movie), HttpStatus.CREATED);
     }
 
     @GetMapping("/{movieId}")
-    public Movie getMovie(@PathVariable String movieId) {
+    public ResponseEntity<Movie> getMovie(@PathVariable String movieId) {
 
-        return movieService.getMovie(movieId);
+        if (!isNumeric(movieId)) {
+            throw new IdNotValidException("you have to provide a valid Id");
+        }
+        if (movieService.getMovie(movieId) == null) {
+            throw new MovieNotFoundException("requested movie not found");
+        }
+
+        try {
+            return new ResponseEntity<>(movieService.getMovie(movieId), HttpStatus.OK);

```

Comment:
You called **movieService.getMovie(movieId)** two times. Each time it's very expensive reading operation. Please save result of the first call in variable and reuse it when you create ResponseEntity.